python/src/com/jetbrains/python/findUsages/PyCommonFindUsageOptions.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2000-2017 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.jetbrains.python.findUsages;
+
+import com.intellij.find.findUsages.FindUsagesHandler;
+import com.intellij.find.findUsages.FindUsagesOptions;
+import com.intellij.openapi.actionSystem.DataContext;
+import com.intellij.openapi.components.PersistentStateComponent;
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.components.State;
+import com.intellij.openapi.project.Project;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * @author Yuli Fiterman
+ */
+@State(name = "PyCommonFindUsageOptions")
+public class PyCommonFindUsageOptions implements PersistentStateComponent<PyCommonFindUsageOptions.State> {
```

Comment:
You can extend `FindUsagesOptions` with your `PyCommonFindUsageOptions` (I would prefer to call it `PyFindUsagesOptions`, we have similarly named classes for Java). Then you'll need only `getInstance()`, `getState()`, and `loadState()` here. The state persisting methods could access `isSearchForTextOccurrences` from the base class.